sdk/core/core-http/src/nodeFetchHttpClient.ts

```diff
@@ -4,21 +4,20 @@
 import * as tough from "tough-cookie";
 import * as http from "http";
 import * as https from "https";
-import "node-fetch";
+import node_fetch from "node-fetch";
 
 import { FetchHttpClient, CommonRequestInfo } from "./fetchHttpClient";
 import { HttpOperationResponse } from "./httpOperationResponse";
 import { WebResourceLike } from "./webResource";
 import { createProxyAgent, ProxyAgent, isUrlHttps } from "./proxyAgent";
 
 interface GlobalWithFetch extends NodeJS.Global {
-  fetch: typeof import("node-fetch")["default"];
+  fetch: typeof node_fetch;
 }
 
 const globalWithFetch = global as GlobalWithFetch;
 if (typeof globalWithFetch.fetch !== "function") {
-  const fetch = require("node-fetch").default;
-  globalWithFetch.fetch = fetch;
+  globalWithFetch.fetch = node_fetch;
```

Comment:
but... why do this? Can't we just use node_fetch directly? This class is only used in node where fetch isn't defined.

Reply:
I actually tried that first, however, the `Request` type defined in node-fetch isn't compatible with `Request` from DOM because it defined additional properties.  `node-fetch` v3.0 doesn't seem to have these.

Can I safely suppress/ignore this error?

```
src/nodeFetchHttpClient.ts:91:23 - error TS2345: Argument of type 'RequestInfo' is not assignable to parameter of type 'import("/home/azuser/git/jssdk/common/temp/node_modules/.pnpm/@types/node-fetch@2.5.7/node_modules/@types/node-fetch/index").RequestInfo'.
  Type 'Request' is not assignable to type 'RequestInfo'.
    Type 'Request' is missing the following properties from type 'Request': context, compress, counter, follow, and 6 more.

91     return node_fetch(input, init);
```

I now realized that current PR still modifies `global` so won't fix the issue.

Reply:
It seems like the trouble is `export type CommonRequestInfo = Request | string;` in `fetchHttpClient` yeah? I think we need to either define our own input interface or re-use node-fetch's

Reply:
Our base class `FetchHttpClient` has this abstract method so we need to have two types that are compatible with both browsers and Node.
```typescript
   abstract async fetch(input: CommonRequestInfo, init?: RequestInit): Promise<Response>;
```
I could narrow down to common things that work for both DOM and NodeJs for `input`: change to `input: string` because it's the only type we ever use. However it seems harder to do similar for `RequestInit` as many things are incompatible.

Reply:
`url: string` is more accurate

Reply:
/cc @willmtemple who made changes around this previously.

Reply:
The returned type `Response` is not compatible either between browser and Node. So we would need three types that are common/intersection of browser and Node.

Reply:
So I got rid of the type errors by defining common types with some of the properties as `any`.  Looks very hacky but not sure whether there are better ways.